app/uk/gov/hmrc/economiccrimelevyenrolment/connectors/EnrolmentStoreProxyConnector.scala

```diff
@@ -40,4 +42,13 @@ class EnrolmentStoreProxyConnectorImpl @Inject() (appConfig: AppConfig, httpClie
       s"$enrolmentStoreUrl/groups/$groupId/enrolments"
     )(readOptionOfNotFoundOrNoContent[GroupEnrolmentsResponse], hc, ec)
 
+  def queryKnownFacts(knownFacts: Seq[KeyValue])(implicit hc: HeaderCarrier): Future[QueryKnownFactsResponse] =
+    httpClient.POST[QueryKnownFactsRequest, QueryKnownFactsResponse](
+      s"$enrolmentStoreUrl/enrolments",
+      QueryKnownFactsRequest(
+        EclEnrolment.ServiceName,
+        knownFacts = knownFacts
+      )
+    )
```

Comment:
I think possibly just need to look at this implementation and see how it's handling a potential 204 response back from the API call (Search did not match any results). From what I can see this would cause an exception to be thrown as there would be no payload to deserialize into the case class.

We do the same as this in the returns frontend but the reason we it's fine if we throw an exception there is because we never expect there to be no results; whereas in the case of this enrolment journey it's a valid scenario to have no results returned